src/renderer/src/routes/Onboarding/CreateProjectScreen.tsx

```diff
@@ -0,0 +1,283 @@
+		defaultMessage: 'Project Name',
+	},
+	addName: {
+		id: 'screens.ProjectCreationScreen.addName',
+		defaultMessage: 'Create Project',
+	},
+	characterCount: {
+		id: 'screens.ProjectCreationScreen.characterCount',
+		defaultMessage: '{count}/{maxLength}',
+	},
+	advancedProjectSettings: {
+		id: 'screens.ProjectCreationScreen.advancedProjectSettings',
+		defaultMessage: 'Advanced Project Settings',
+	},
+	importConfig: {
+		id: 'screens.ProjectCreationScreen.importConfig',
+		defaultMessage: 'Import Config',
+	},
+	errorSavingProjectName: {
+		id: 'screens.ProjectCreationScreen.errorSavingProjectName',
+		defaultMessage:
+			'An error occurred while saving your project name. Please try again.',
+	},
+	saving: {
+		id: 'screens.ProjectCreationScreen.saving',
+		defaultMessage: 'Saving...',
+	},
+})
+
+export const Route = createFileRoute('/Onboarding/CreateProjectScreen')({
+	component: CreateJoinProjectScreenComponent,
+})
+
+const StyledImage = styled('img')({
+	width: 60,
+	height: 60,
+})
+
+const InputWrapper = styled('div')({
+	marginTop: 24,
+	marginBottom: 24,
+	display: 'flex',
+	flexDirection: 'column',
+	alignItems: 'center',
+})
+
+const StyledTextField = styled(TextField)({
+	width: '100%',
+	maxWidth: 400,
+	backgroundColor: WHITE,
+	marginTop: 20,
+})
+
+const CharacterCount = styled(Text, {
+	shouldForwardProp: (prop) => prop !== 'error',
+})<{ error: boolean }>(({ error }) => ({
+	marginTop: 8,
+	color: error ? RED : BLACK,
+	width: '100%',
+	maxWidth: 400,
+	textAlign: 'right',
+}))
+
+const HorizontalLine = styled('div')({
+	borderBottom: `1px solid ${BLUE_GREY}`,
+	margin: '20px auto 20px auto',
+	width: '65%',
+})
+
+function CreateJoinProjectScreenComponent() {
+	const navigate = useNavigate()
+	const { formatMessage } = useIntl()
+	const [projectName, setProjectName] = useState('')
+	const [error, setError] = useState(false)
+	const [errorMessage, setErrorMessage] = useState('')
+	const setProjectNameMutation = useCreateProject()
+
+	const [advancedSettingOpen, setAdvancedSettingOpen] = useState(false)
+	const [configFileName, setConfigFileName] = useState<string | null>(null)
+
+	function countGraphemes(text: string): number {
+		const segmenter = new Intl.Segmenter(undefined, {
+			granularity: 'grapheme',
+		})
+		let result = 0
+		for (const _ of segmenter.segment(text)) result++
+		return result
+	}
+
+	function getUtf8ByteLength(text: string): number {
+		return new TextEncoder().encode(text).length
+	}
+
+	const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+		const value = event.target.value
+		const graphemeCount = countGraphemes(value.trim())
+		const byteLength = getUtf8ByteLength(value.trim())
+		let localError = false
+
+		if (
+			graphemeCount > PROJECT_NAME_MAX_LENGTH_GRAPHEMES ||
+			byteLength > PROJECT_NAME_MAX_BYTES ||
+			value.trim().length === 0
+		) {
+			localError = true
+		} else {
+			setProjectName(value)
+		}
+		setError(localError)
```

Comment:
can we write a test for this?

we could write a pure function that does not deal with react state

Something like:

```ts
const handleChange = (event: ChangeEvent<HTMLInputElement>) =>{
     const value = event.target.value
     const localError:boolean= checkForError(value)
     if (!localError) setProjectName(value)
     if (localError !== error) setError(localError)
}
```

then we could write a test for `checkForError` which is not reliant on react state, and can be a pure function

Reply:
I think I did this. I tried to anyway.